assembly/UInt/UInt128.ts

```diff
@@ -62,29 +57,61 @@ export class UInt128 implements Codec {
     toString(): string {
         return this.value.toString();
     }
-
+    /**
+     * @description Non-static constructor method used to populate defined properties of the model
+     * @param bytes SCALE encoded bytes
+     * @param index index to start decoding the bytes from
+     */
+    populateFromBytes(bytes: u8[], index: i32 = 0): void{
+        assert(bytes.length - index > 0, 'Invalid input: Byte array should not be empty');
+        const value = UInt128._computeValue(bytes.slice(index));
+        this._value = value;
+        this.bitLength = UInt128._computeBitLength(this._value);
+    }
     /**
      * @description The length of Int when the value is encoded
      */
     public encodedLength (): i32 {
         return this.bitLength;
     }
 
-    /** Instantiates new UInt128 from u8[] SCALE encoded bytes */
-    static fromU8a(input: u8[]): UInt128 {
-        assert(input.length != 0, 'Invalid input: Byte array should not be empty');
-        const mode = input[0] & 0x03;
+    /**
+     * Internal static private function to compute value of the UInt128
+     * @param bytes 
+     * @param index 
+     */
+    static _computeValue(bytes: u8[]): u128{
```

Comment:
I guess that we could have the additional `index` property. This way we won't have to slice the array when we do `populateFromBytes` or `fromU8a`